bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/AndroidTVHandler.java

```diff
@@ -133,11 +136,15 @@ public void checkThingStatus() {
             }
         }
 
-        if (failed) {
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE, statusMessage);
-        } else {
-            updateStatus(ThingStatus.ONLINE);
+        if (!currentThingStatus.equals(statusMessage)) {
+            if (failed) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE, statusMessage);
+            } else {
+                updateStatus(ThingStatus.ONLINE);
```

Comment:
There is now a high risk that you thing status is now never set to ONLINE.

Reply:
Can you elaborate on that?  I'm not seeing the risk.  At some point in the cycle, the status message will be changed to just ONLINE which will make it through the first conditional, and then since failed is still false it will get marked as online.

Reply:
Example:

```
2023-07-17 09:19:10.911 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'androidtv:shieldtv:livingroom' changed from INITIALIZING to OFFLINE: Protocols Starting
2023-07-17 09:19:10.943 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'androidtv:shieldtv:livingroom' changed from OFFLINE: Protocols Starting to OFFLINE: GoogleTV:  | ShieldTV: Initializing
2023-07-17 09:19:10.957 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'androidtv:shieldtv:livingroom' changed from OFFLINE: GoogleTV:  | ShieldTV: Initializing to OFFLINE: GoogleTV: Initializing | ShieldTV: Initializing
2023-07-17 09:19:11.003 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'androidtv:shieldtv:livingroom' changed from OFFLINE: GoogleTV: Initializing | ShieldTV: Initializing to OFFLINE: GoogleTV: Online | ShieldTV: Initializing
2023-07-17 09:19:11.025 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'androidtv:shieldtv:livingroom' changed from OFFLINE: GoogleTV: Online | ShieldTV: Initializing to ONLINE

```

In the end, both GTV and STV change their status to Online, which should change currentThingStatus.  At that point, since failed is false, the thing goes online.

Reply:
I mean currentThingStatus is initialized to "". In case the first time you run the method, statusMessage is also "", your if condition will be wrong and as a result you will not set the thing status (to ONLINE).

Reply:
Why not simply initializing your variable with a value that cannot happen to be sure to enter in the if the first time ?
That pattern code looks not really good to me.
But as you say it works and you seems to be sure of you, let's merge it as it is...